pycarol/bigquery.py

```diff
@@ -265,37 +268,49 @@ def query(
             dataset_id: BigQuery dataset ID, if not provided, it will use the default
                         one.
             return_dataframe: Return dataframe if True.
+            return_job_id : If True, returns an tuple containing the query results with the job-id on BigQuery platform.
+            retry: Custom google.api_core.retry.Retry object to adjust Google`s BigQuery API
+                   calls, to custom timeout and exceptions to retry.
 
         Returns:
             Query result.
 
         Usage:
-
         .. code:: python
 
 
             from pycarol import BQ, Carol
 
             bq = BQ(Carol())
             query = 'select * from invoice limit 10'
+            
+            #Normal use
             df = bq.query(query, return_dataframe=True)
+
+            #Custom retry object
+            from google.api_core.retry import Retry
+            df = bq.query(query, return_dataframe=True, retry=Retry(initial=2, multiplier=2, maximum=60, timeout=200))
+
+            #Getting BigQuery`s Job-id (Util for debugging in platform)
+            df, job_id_string = bq.query(query, return_dataframe=True, return_job_id=True)
+
         """
         service_account = self._token_manager.get_token().service_account
         client = self._generate_client(service_account)
 
         dataset_id = dataset_id or self._dataset_id
         job_config = bigquery.QueryJobConfig(default_dataset=dataset_id)
-        results_job = client.query(query, job_config=job_config)
+        results_job = client.query(query, retry=retry, job_config=job_config) if retry else client.query(query, job_config=job_config)
```

Comment:
do you need this? 
Retry will be None, so this will not change the behaviour. 
if retry is None 
`client.query(query, retry=retry, job_config=job_config)` and `client.query(query, job_config=job_config)` are the same

Reply:
No, in google.cloud.bigquery Client class method query(), thats pyCarol use, if retry=None then it will not set any retry logic, so thre's different from passing retry=None and not passing any parameter.

Reply:
Just for future reference that retry default is already the Retry object 
[doc](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query)